File: Java/Medium/WrongAnswer/partition-array-according-to-given-pivot_9.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        
        int n = nums.length;
        int[] ans = new int[n];
        int i = 0;
        int j = 0;
        int k = n - 1;
        while (j <= k) {
            if (nums[j] < pivot) {
                ans[i] = nums[j];
                j++;
                i++;
            } else if (nums[j] == pivot) {
                j++;
            } else {
                ans[k] = nums[j];
                k--;
            }
        }
        return ans;
    }
}

File: Rust/Medium/WrongAnswer/partition-array-according-to-given-pivot_9.rs
impl Solution {
    pub fn pivot_array(nums: Vec<i32>, pivot: i32) -> Vec<i32> {
        
        let mut res = Vec::new();
        let mut mid = Vec::new();
        for num in nums {
            if num < pivot {
                res.push(num);
            } else if num > pivot {
                mid.push(num);
            } else {
                res.push(num);
            }
        }
        res.append(&mut mid);
        res
    }
}